packages/commons/src/auth/jwt.ts

```diff
@@ -47,6 +60,52 @@ export const readAuthDataFromJwtToken = (
   }
 };
 
+const clients = !config.skipJWTVerification
+  ? config.wellKnownUrls.map((url) =>
+      jwksClient({
+        jwksUri: url,
+      })
+    )
+  : undefined;
+
+const getKey =
+  (
+    clients: jwksClient.JwksClient[]
+  ): ((header: JwtHeader, callback: SigningKeyCallback) => void) =>
+  (header, callback) => {
+    for (const { client, last } of clients.map((c, i) => ({
+      client: c,
+      last: i === clients.length - 1,
+    }))) {
+      client.getSigningKey(header.kid, function (err, key) {
+        if (err && last) {
+          logger.error(`Error getting signing key: ${err}`);
+          return callback(err, undefined);
+        } else {
+          return callback(null, key?.getPublicKey());
+        }
+      });
+    }
+  };
+
+export const verifyJwtToken = (jwtToken: string): Promise<boolean> =>
+  clients === undefined
+    ? Promise.resolve(true)
+    : new Promise((resolve, _reject) => {
+        jwt.verify(
+          jwtToken,
+          getKey(clients),
+          undefined,
+          function (err, _decoded) {
+            if (err) {
+              logger.error(`Error verifying token: ${err}`);
```

Comment:
may I ask you to verify what the `err` contains?
We must to be sure that the token is not leaked into logs, even if not valid

Reply:
here: https://www.npmjs.com/package/jsonwebtoken?activeTab=readme there's a section `Errors & Codes` with descriptions of all possible problems.
But we also could simply remove `err` from the log and leave only `Error verifying token`

Reply:
Having the rejection reason in the log is very useful, if `err` will contain only the messages in the documentation, it's a go for me ✅